admin/react/assets/src/dashboard-app/pages/fonts/preview/GFontVariation.js

```diff
@@ -0,0 +1,135 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { __ } from "@wordpress/i18n";
+
+const GFontVariation = (props) => {
+	const { weight, font, isInGoogleState } = props;
+	const googleFont = useSelector( ( state ) => state.googleFont );
+	const dispatch = useDispatch();
+
+	const getFontWeightTitle = ( weight ) => {
+		switch ( weight ) {
+			case '100':
+				return __( 'Thin ', 'custom-fonts' ) + weight;
+			case '200':
+				return __( 'Extra Light ', 'custom-fonts' ) + weight;
+			case '300':
+				return __( 'Light ', 'custom-fonts' ) + weight;
+			case '400':
+				return __( 'Regular ', 'custom-fonts' ) + weight;
+			case '500':
+				return __( 'Medium ', 'custom-fonts' ) + weight;
+			case '600':
+				return __( 'Semi Bold ', 'custom-fonts' ) + weight;
+			case '700':
+				return __( 'Bold ', 'custom-fonts' ) + weight;
+			case '800':
+				return __( 'Extra Bold ', 'custom-fonts' ) + weight;
+			case '900':
+				return __( 'Black ', 'custom-fonts' ) + weight;
+			default:
+				return __( 'Weight ', 'custom-fonts' ) + weight;
+		}
+	}
+
+	const addWeight = (e) => {
+		e.preventDefault();
+		e.stopPropagation();
+
+		const varWt = e.target.dataset.font_weight;
+		const variations = googleFont.variations;
+		if ( undefined === varWt ) {
+			return;
+		}
+		variations.push( {
+			id: variations.length+1,
+			font_file: '',
+			font_style: '',
+			font_weight: varWt
+		} );
+		dispatch( { type: 'SET_GOOGLE_FONT', payload: {
+			"font_name": googleFont.font_name,
+			"font_fallback": googleFont.font_fallback,
+			"font_display": googleFont.font_display,
+			"variations": variations
+		} } );
+	}
+
+	const removeWeight = (e) => {
+		e.preventDefault();
+		e.stopPropagation();
+
+		const updatedVariations = googleFont.variations.filter(
+			(variation) => variation.font_weight !== weight
+		);
+
+		dispatch( { type: 'SET_GOOGLE_FONT', payload: {
+			"font_name": googleFont.font_name ? googleFont.font_name : '',
+			"font_fallback": googleFont.font_fallback ? googleFont.font_fallback : '',
+			"font_display": googleFont.font_display ? googleFont.font_display : '',
+			"variations": updatedVariations
+		} } );
+	}
+
+	const getRenderFontWeight = (weight) => {
+		return weight.replace( "italic", "" );
+	}
+
+	return (
+		<div className="py-5">
+			<div className="flex justify-between items-center">
+				<div>
+					{/* Variation Name */}
+					<div className="text-sm text-neutral mb-3.5">
+						{ getFontWeightTitle(weight) }
+					</div>
+					{/* Variation Preview */}
+					<div className="text-5xl" style={{ fontFamily: font, fontWeight: getRenderFontWeight(weight), fontSize: "var(--bsf-custom-font-size)" }}>
+						{__('How vexingly quick daft zebras jump!', 'custom-fonts')}
+					</div>
+				</div>
+				<div>
+					{ ( ! isInGoogleState ) &&
+						<button className="flex text-sm text-primary items-center py-2 px-3 border border-primary" data-font_weight={weight} onClick={addWeight}>
+							<svg
```

Comment:
Same here @imnavanath regarding svg reuse

Reply:
Added separate task in backlog for this - https://brainstormforce.atlassian.net/browse/CFP-38